sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/HybridSearchDocumentQueryExecutionContext.java

```diff
@@ -302,7 +307,7 @@ private static Flux<HybridSearchQueryResult<Document>> computeRRFScores(Mono<Lis
                 for (int index = 0; index < results.size(); ++index) {
                     double rrfScore = 0.0;
                     for (List<Integer> integers : ranksInternal) {
-                        rrfScore += 1.0 / (RRF_CONSTANT + integers.get(index));
+                        rrfScore += componentWeights.get(index).getWeight() / (RRF_CONSTANT + integers.get(index));
```

Comment:
possiblity of IndexOutOfBounds exception?

Reply:
This code can not go into this exception because when we are creating the componentWeights list, backend will throw an error if there is a size mismatch in the length of component score list and component weight list.

Reply:
Do we have a test case to verify the exception?

Reply:
Added the test case for that.